internal/services/elastic/elastic_tag_rule_resource.go

```diff
@@ -0,0 +1,308 @@
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ElasticMonitorName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"rule_set_name": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  utils.String("default"),
+			},
+
+			"id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"log_rules": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"send_aad_logs": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"send_subscription_logs": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"send_activity_logs": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"filtering_tag": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"name": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+									},
+									"value": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+									},
+									"action": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"Include",
+											"Exclude",
+										}, false),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"provisioning_state": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceElasticTagRuleCreateorUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Elastic.TagRuleClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	ruleSetName := d.Get("rule_set_name").(string)
+
+	id := parse.NewElasticTagRuleID(subscriptionId, resourceGroup, name, ruleSetName).ID()
+
+	existing, err := client.Get(ctx, resourceGroup, name, ruleSetName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing Elastic Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	body := elastic.MonitoringTagRules{
+		Properties: &elastic.MonitoringTagRulesProperties{
+			LogRules: expandLogRules(d.Get("log_rules").([]interface{})),
+		},
+	}
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, ruleSetName, &body); err != nil {
+		return fmt.Errorf("configuring Tag Rules on Elastic Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
```

Comment:
The log can be simplified by using id as a parameter.

Reply:
updated